pkg/dogstatsd/server_test.go

```diff
@@ -56,7 +56,8 @@ func TestNewServer(t *testing.T) {
 
 	demux := aggregator.InitTestAgentDemultiplexerWithFlushInterval(10 * time.Millisecond)
 	defer demux.Stop(false)
-	s, err := NewServer(demux, false)
+	s := NewServer(false)
+	_ = s.Start(demux)
 	require.NoError(t, err, "cannot start DSD")
```

Comment:
`require.NoError` now checks an old value of err. Would it make sense to check the value from `s.Start` instead? Same pattern repeats in most other tests.